src/units.cpp

```diff
@@ -410,11 +410,17 @@ size_t Units::unitCount() const
 double Units::scalingFactor(const UnitsPtr &units1, const UnitsPtr &units2)
 {
     double multiplier = 0.0;
+    double scalingFactor = 0.0;
 
-    updateUnitMultipliers(multiplier, units2, 1, 0, 1);
-    updateUnitMultipliers(multiplier, units1, 1, 0, -1);
+    if (!units1 || !units2) {
+        // no change
+    } else {
+        updateUnitMultipliers(multiplier, units2, 1, 0, 1);
+        updateUnitMultipliers(multiplier, units1, 1, 0, -1);
+        scalingFactor = std::pow(10, multiplier);
+    }
 
-    return std::pow(10, multiplier);
+    return scalingFactor;
 }
```

Comment:
I would simplify that method so that it ends up reading something like:

```
double Units::scalingFactor(const UnitsPtr &units1, const UnitsPtr &units2)
{
    if ((units1 != nullptr) && (units2 != nullptr)) {
        double multiplier = 0.0;

        updateUnitMultipliers(multiplier, units1, 1, 0, -1);
        updateUnitMultipliers(multiplier, units2, 1, 0, 1);

        return std::pow(10, multiplier);
    }

    return 0.0;
}
```

Reply:
I see, that's way cleaner than what I had before. I'll make that change now.